src/ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp

```diff
@@ -0,0 +1,82 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <tuple>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/FixedHashMap.hpp"
+#include "DataStructures/Index.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+
+namespace LinearSolver::Schwarz {
+
+/// Identifies a subdomain region that overlaps with another element
+template <size_t Dim>
+using OverlapId = std::pair<Direction<Dim>, ElementId<Dim>>;
+
+/// Data structure that can store the `ValueType` on each possible overlap of an
+/// element-centered subdomain with its neighbors. Overlaps are identified by
+/// their `OverlapId`.
+template <size_t Dim, typename ValueType>
+using OverlapMap =
+    FixedHashMap<maximum_number_of_neighbors(Dim), OverlapId<Dim>, ValueType,
+                 boost::hash<OverlapId<Dim>>>;
+
+/*!
+ * \brief The number of points that an overlap extends into the `volume_extent`
```

Comment:
What's not clear to me is whether this is the number of points in the volume sense or the number of points in the normal direction to the share face. Could you clarify that in the comment?